packages/serverpod/lib/src/database/database_query.dart

```diff
@@ -61,16 +68,19 @@ class SelectQueryBuilder {
       countTableRelation: _countTableRelation,
     );
 
-    var join = _buildJoinQuery(
-      where: _where,
-      manyRelationWhereAddition: _manyRelationWhereAddition,
-      having: _having,
-      orderBy: _orderBy,
-      include: _include,
-      subQueries: subQueries,
-      countTableRelation: _countTableRelation,
-      joinOneLevelManyRelations: _joinOneLevelManyRelationWhereExpressions,
-    );
+    String? join;
+    if (!_viewTable) {
+      join = _buildJoinQuery(
+        where: _where,
+        manyRelationWhereAddition: _manyRelationWhereAddition,
+        having: _having,
+        orderBy: _orderBy,
+        include: _include,
+        subQueries: subQueries,
+        countTableRelation: _countTableRelation,
+        joinOneLevelManyRelations: _joinOneLevelManyRelationWhereExpressions,
+      );
+    }
```

Comment:
Suggestion: Add a throw if a join is attempted while a view table is used.

This will silently not join a table if the view table property is set to true. That probably won't yield the result the developer is after and it is better to inform them that this operation is not supported.